modules/apps/polls/polls-service/src/main/java/com/liferay/polls/service/impl/PollsQuestionLocalServiceImpl.java

```diff
@@ -265,29 +260,6 @@ public int getQuestionsCount(long groupId) {
 		long companyId, long[] groupIds, String keywords, int start, int end,
 		OrderByComparator<PollsQuestion> orderByComparator) {
 
-		try {
-			Hits hits = searchIndexer(
-				companyId, groupIds, keywords, start, end, orderByComparator);
-
-			List<PollsQuestion> pollsQuestions = new ArrayList<>(
-				hits.getLength());
-
-			for (Document document : hits.getDocs()) {
-				Long questionId = GetterUtil.getLong(
-					document.get(Field.ENTRY_CLASS_PK));
-
-				pollsQuestions.add(
-					pollsQuestionPersistence.fetchByPrimaryKey(questionId));
-			}
-
-			return pollsQuestions;
-		}
-		catch (PortalException portalException) {
-			if (_log.isDebugEnabled()) {
-				_log.debug(portalException, portalException);
-			}
-		}
-
```

Comment:
Hi @natocesarrego, I removed this code because, after some tests, I was getting the following error: “_Polls is temporarily unavailable_”. 
The reason why is because the `searchIndexer` method was returning _indexes_ (`hits.getLength() > 0`) even when I had no polls. This fact was causing the `pollsQuestions` variable to be a list with a _null_ element (`[null]`). 
Because of this unstable behavior, and because I was using `DDMFormInstanceServiceImpl.java` as an example, I decided to remove it.
But now I realize that this bug can also be fixed if I check that `pollsQuestionPersistence.fetchByPrimaryKey(questionId)` is null before adding.
What do you think is the best alternative?

Reply:
Thanks for your explanation, @carolmariaabb. 😉 First of all, we should investigate what is populating the `hits` and with which content, because this is the main point causing the issue. But, I think your suggestion is good too. Can you please investigate what is populating the `hits` and re-send? Thanks.